code/modules/mapping/map_template.dm

```diff
@@ -108,19 +176,36 @@
 	parsed.initTemplateBounds()
 
 	log_game("[name] loaded at [T.x],[T.y],[T.z]")
-	return bounds
+	on_map_loaded(T.z, parsed.bounds)
 
-/datum/map_template/proc/get_affected_turfs(turf/T, centered = FALSE)
-	var/turf/placement = T
-	if(centered)
-		var/turf/corner = locate(placement.x - round(width/2), placement.y - round(height/2), placement.z)
-		if(corner)
-			placement = corner
-	return block(placement, locate(placement.x+width-1, placement.y+height-1, placement.z))
+	return bounds
 
+//This, get_affected_turfs, and load() calculations for bounds/center can probably be optimized. Later.
+/datum/map_template/proc/annihilate_bounds(turf/origin, centered = FALSE, orientation = SOUTH)
+	var/deleted_atoms = 0
+	log_world("Annihilating objects in map loading location.")
+	var/list/turfs_to_clean = get_affected_turfs(origin, centered, orientation)
+	if(turfs_to_clean.len)
+		var/list/kill_these = list()
+		for(var/i in turfs_to_clean)
+			var/turf/T = i
+			kill_these += T.contents
+		for(var/i in kill_these)
+			qdel(i)
+			CHECK_TICK
+			deleted_atoms++
+	log_world("Annihilated [deleted_atoms] objects.")
 
 //for your ever biggening badminnery kevinz000
 //❤ - Cyberboss
-/proc/load_new_z_level(file, name, list/traits)
+/proc/load_new_z_level(file, name, orientation, list/ztraits)
 	var/datum/map_template/template = new(file, name)
-	return template.load_new_z(traits)
+	template.load_new_z(orientation, ztraits)
```

Comment:
```suggestion
	return template.load_new_z(orientation, ztraits)
```
The return value is used in createRandomZlevel to announce wheter the zlevel was successfully loaded or not.